pkg/tcpip/network/ip_test.go

```diff
@@ -160,14 +173,16 @@ func (t *testObject) WritePacket(_ *stack.Route, _ *stack.GSO, hdr buffer.Prepen
 		prot = tcpip.TransportProtocolNumber(h.Protocol())
 		srcAddr = h.SourceAddress()
 		dstAddr = h.DestinationAddress()
+		t.checkValues(prot, payload, srcAddr, dstAddr, options)
 
 	} else {
 		h := header.IPv6(hdr.View())
 		prot = tcpip.TransportProtocolNumber(h.NextHeader())
 		srcAddr = h.SourceAddress()
 		dstAddr = h.DestinationAddress()
+		options = nil // Because we are not going to check options in IPv6 packets
+		t.checkValues(prot, payload, srcAddr, dstAddr, options)
 	}
```

Comment:
Just pass nil for options?

Reply:
while testing ipv4 options we set should not affect while testing ipv6, that's why passing nil.